static/js/ProfileDetail.tsx

```diff
@@ -0,0 +1,232 @@
+import { fetchProfile } from "./api/profiles";
+import DeleteProfileBtn from "./buttons/profiles/DeleteProfileBtn";
+import EditProfileBtn from "./buttons/profiles/EditProfileBtn";
+import OpenProfileListBtn from "./buttons/profiles/OpenProfileListBtn";
+import BaseLayout from "./components/BaseLayout";
+import NotificationRow from "./components/NotificationRow";
+import { queryKeys } from "./util/queryKeys";
+import useNotification from "./util/useNotification";
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Col,
+  Row,
+} from "@canonical/react-components";
+import { LxdDiskDevice, LxdNicDevice } from "./types/device";
+import { isDiskDevice, isNicDevice } from "./util/type";
+
+const ProfileDetail: FC = () => {
+  const { name } = useParams<{ name: string }>();
+
+  if (!name) {
+    return <>Missing name</>;
+  }
+
+  const notify = useNotification();
+
+  const {
+    data: profile,
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.profiles, name],
+    queryFn: () => fetchProfile(name),
+  });
+
+  if (error) {
+    notify.failure("Could not load profile details.", error);
+  }
+
+  if (isLoading) {
+    return <>Loading...</>; // TODO: improve, maybe with a nice loader?
+  } else if (!profile) {
+    return <>Could not load profile details.</>;
+  }
+
+  const btnProps = {
+    profile: profile,
+    notify: notify,
+    appearance: "",
+    className: "u-no-margin--bottom",
+    isDense: false,
+  };
+
+  const getUsedByNames = () => {
+    return profile.used_by?.map((path) => path.split("/").slice(-1)[0]);
+  };
+
+  const getCloudInitConfig = (
+    type: "user-data" | "vendor-data" | "network-config"
+  ) => {
+    return profile.config[`cloud-init.${type}`].replace("|\n", "");
+  };
+
+  return (
+    <BaseLayout
+      title={`Profile details for ${name}`}
+      controls={
+        <>
+          <EditProfileBtn label="Edit" {...btnProps} />
+          {profile.name !== "default" && (
+            <DeleteProfileBtn label="Delete" name={name} {...btnProps} />
+          )}
+          <OpenProfileListBtn />
+        </>
+      }
+    >
+      <NotificationRow notify={notify} />
+      <Row>
+        <table>
+          <thead>
+            <tr>
+              <th>Label</th>
+              <th>Value</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <th>Name</th>
+              <td>{profile.name}</td>
+            </tr>
+            <tr>
+              <th>Description</th>
+              <td>{profile.description}</td>
+            </tr>
+            <tr>
+              <th>Instances using this profile</th>
+              <td>{getUsedByNames()?.join(", ")}</td>
+            </tr>
+            {Object.values(profile.devices).some((device) =>
+              isNicDevice(device)
+            ) && (
+              <tr>
+                <th>Network devices</th>
+                <td>
+                  {Object.values(profile.devices)
```

Comment:
In my opinion, it is better to always show the rows with headings and not filter them out if there is no content. The content in that case can be "none" or just "-". The reason I think this is better, is users remembering the place where to find an information if the row with heading is always there. If the row is sometimes there and sometimes not, users tend to be confused or at least they don't get the mental model of finding this information in that place as fast as in the other approach.

Reply:
Sure, good point. I will fix that. I am also already using this approach for the `used_by` information after all.

Reply:
Fixed @ [3b362fd](https://github.com/canonical/lxd-ui/commit/3b362fdf2a1568565b63d47c47b20775395fa671)